modules/vaos/app/services/vaos/preferences_service.rb

```diff
@@ -11,12 +11,36 @@ class PreferencesService < Common::Client::Base
 
     STATSD_KEY_PREFIX = 'api.vaos'
 
-    def get_preferences(user)
+    attr_reader :user
+
+    def initialize(user)
+      @user = user
+    end
+
+    def get_preferences
       with_monitoring do
-        url = "/var/VeteranAppointmentRequestService/v4/rest/patient/ICN/#{user.icn}/preference"
         response = perform(:get, url, nil, headers(user))
-        OpenStruct.new(response.body.merge(id: SecureRandom.uuid))
+        OpenStruct.new(response.body.merge(id: preference_id))
```

Comment:
Any chance this would return an error worth `rescue`-ing? Ditto for the `put_preferences` action?

Reply:
no, it would raise an exception in the JSON middleware before it raises an exception here.